### ISSUE TITLE
zammad-postgresql_1 / zammad-init_1 FATAL:  role "zammad" does not exist

### ISSUE THREAD

**Issue by Waasil** [zammad/zammad-docker-compose#133]
### Infos:

Hello. I have problem with zammad-docker-compose. Depends on zammad-docker-compose version I try to use I had different problem so I have decided to split them into **Problem_v1** and **Problem_v2**.

* Docker version:
 Version:           19.03.6
 API version:       1.40
 Go version:        go1.12.16
 Git commit:        369ce74a3c
 Built:             Thu Feb 13 01:27:49 2020
 OS/Arch:           linux/amd64
 Experimental:      false

* Docker-compose version:
docker-compose version 1.17.1, build unknown
docker-py version: 2.5.1
CPython version: 2.7.15+
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018

* Operating system (Docker host):
Ubuntu 18.04.3 LTS 

### Expected behavior:

Installation without errors

### Actual behavior:

```
zammad-elasticsearch_1  | [2020-02-25T15:30:26,158][INFO ][o.e.n.Node               ] [] initializing ...
zammad-elasticsearch_1  | [2020-02-25T15:30:26,325][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [] uncaught exception in thread [main]
zammad-elasticsearch_1  | org.elasticsearch.bootstrap.StartupException: ElasticsearchException[java.io.IOException: failed to read [id:19, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/node-19.st]]; nested: IOException[failed to read [id:19, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/node-19.st]]; nested: IllegalArgumentException[[node_meta_data] unknown field [node_version], parser not found];
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:136) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:123) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:70) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | Caused by: org.elasticsearch.ElasticsearchException: java.io.IOException: failed to read [id:19, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/node-19.st]
zammad-elasticsearch_1  | 	at org.elasticsearch.ExceptionsHelper.maybeThrowRuntimeAndSuppress(ExceptionsHelper.java:196) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.gateway.MetaDataStateFormat.loadLatestState(MetaDataStateFormat.java:335) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeEnvironment.loadOrCreateNodeMetaData(NodeEnvironment.java:418) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:267) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.node.Node.<init>(Node.java:265) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.node.Node.<init>(Node.java:245) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:233) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:233) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:342) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:132) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	... 6 more
zammad-elasticsearch_1  | Caused by: java.io.IOException: failed to read [id:19, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/node-19.st]
zammad-elasticsearch_1  | 	at org.elasticsearch.gateway.MetaDataStateFormat.loadLatestState(MetaDataStateFormat.java:328) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeEnvironment.loadOrCreateNodeMetaData(NodeEnvironment.java:418) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:267) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.node.Node.<init>(Node.java:265) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.node.Node.<init>(Node.java:245) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:233) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:233) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:342) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:132) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	... 6 more
zammad-elasticsearch_1  | Caused by: java.lang.IllegalArgumentException: [node_meta_data] unknown field [node_version], parser not found
zammad-elasticsearch_1  | 	at org.elasticsearch.common.xcontent.ObjectParser.getParser(ObjectParser.java:399) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:159) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.common.xcontent.ObjectParser.apply(ObjectParser.java:183) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeMetaData$1.fromXContent(NodeMetaData.java:110) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeMetaData$1.fromXContent(NodeMetaData.java:94) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:203) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.gateway.MetaDataStateFormat.loadLatestState(MetaDataStateFormat.java:323) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeEnvironment.loadOrCreateNodeMetaData(NodeEnvironment.java:418) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:267) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.node.Node.<init>(Node.java:265) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.node.Node.<init>(Node.java:245) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:233) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:233) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:342) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:132) ~[elasticsearch-5.6.14.jar:5.6.14]
zammad-elasticsearch_1  | 	... 6 more
zammad-scheduler_1      | waiting for init container to finish install or update...
zammad-init_1           | zammad railsserver waiting for elasticsearch server to be ready...
```

### Steps to reproduce the behavior:

* I have changed version in `.env` file into VERSION = -3.2.0-12 
( I have tried also -3.2.0-11 but result was the same)
* I have increased sysctl -w vm.max_map_count to 327680
* I have changed privileges of path `/usr/share` into 777 based on #102 issue
* Command `docker-compose up`

**Summary**
Should I install Elasticsearch before `docker-compose up` or docker-compose should provide me that service?

### Expected behavior:

Installation without errors

### Actual behavior:
```
zammad-elasticsearch_1  | OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:38,731Z", "level": "INFO", "component": "o.e.e.NodeEnvironment", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/mapper/ubuntu--vg-ubuntu--lv)]], net usable_space [12gb], net total_space [48.1gb], types [ext4]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:38,738Z", "level": "INFO", "component": "o.e.e.NodeEnvironment", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "heap size [990.7mb], compressed ordinary object pointers [true]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:39,063Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "node name [0f0e97a8c9e6], node ID [QQZgyu48SHWMiyIYVxn8oQ], cluster name [docker-cluster]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:39,064Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "version[7.6.0], pid[1], build[default/docker/7f634e9f44834fbc12724506cc1da681b0c3b1e3/2020-02-06T00:09:00.449973Z], OS[Linux/4.15.0-88-generic/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/13.0.2/13.0.2+8]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:39,064Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "JVM home [/usr/share/elasticsearch/jdk]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:39,065Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=COMPAT, -Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Djava.io.tmpdir=/tmp/elasticsearch-11272749202282383895, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Des.cgroups.hierarchy.override=/, -XX:MaxDirectMemorySize=536870912, -Des.path.home=/usr/share/elasticsearch, -Des.path.conf=/usr/share/elasticsearch/config, -Des.distribution.flavor=default, -Des.distribution.type=docker, -Des.bundled_jdk=true]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,735Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [aggs-matrix-stats]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,736Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [analysis-common]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,736Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [flattened]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,737Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [frozen-indices]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,737Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [ingest-common]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,738Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [ingest-geoip]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,745Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [ingest-user-agent]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,746Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [lang-expression]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,746Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [lang-mustache]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,746Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [lang-painless]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,746Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [mapper-extras]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,747Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [parent-join]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,747Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [percolator]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,748Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [rank-eval]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,748Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [reindex]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,749Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [repository-url]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,750Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [search-business-rules]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,752Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [spatial]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,752Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [transform]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,752Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [transport-netty4]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,754Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [vectors]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,755Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-analytics]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,755Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-ccr]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,755Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-core]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,756Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-deprecation]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,756Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-enrich]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,756Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-graph]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,756Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-ilm]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,757Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-logstash]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,758Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-ml]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,758Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-monitoring]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,759Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-rollup]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,760Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-security]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,760Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-sql]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,761Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-voting-only-node]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,761Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded module [x-pack-watcher]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:41,762Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "loaded plugin [ingest-attachment]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:48,669Z", "level": "INFO", "component": "o.e.x.s.a.s.FileRolesStore", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "parsed [0] roles from file [/usr/share/elasticsearch/config/roles.yml]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:49,472Z", "level": "INFO", "component": "o.e.x.m.p.l.CppLogMessageHandler", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "[controller/117] [Main.cc@110] controller (64 bit): Version 7.6.0 (Build 1c8cca13fa9631) Copyright (c) 2020 Elasticsearch BV" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:50,093Z", "level": "DEBUG", "component": "o.e.a.ActionModule", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "Using REST wrapper from plugin org.elasticsearch.xpack.security.Security" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:50,257Z", "level": "INFO", "component": "o.e.d.DiscoveryModule", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "using discovery type [single-node] and seed hosts providers [settings]" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:51,687Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "initialized" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:51,687Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "starting ..." }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:51,841Z", "level": "INFO", "component": "o.e.t.TransportService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "publish_address {192.168.32.3:9300}, bound_addresses {0.0.0.0:9300}" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:52,524Z", "level": "INFO", "component": "o.e.c.c.Coordinator", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "cluster UUID [I4fAGa7GRXCijfDjrvdiNw]" }
zammad-postgresql_1     | 
zammad-init_1           | initialising / updating database...
zammad-backup_1         | waiting for railsserver to be ready...
zammad-init_1           | creating database...
zammad-postgresql_1     | PostgreSQL Database directory appears to contain a database; Skipping initialization
zammad-postgresql_1     | 
zammad-postgresql_1     | LOG:  database system was shut down at 2020-02-25 15:31:07 UTC
zammad-postgresql_1     | LOG:  MultiXact member wraparound protections are now enabled
zammad-postgresql_1     | LOG:  database system is ready to accept connections
zammad-postgresql_1     | LOG:  autovacuum launcher started
zammad-postgresql_1     | LOG:  incomplete startup packet
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:52,775Z", "level": "INFO", "component": "o.e.c.s.MasterService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "elected-as-master ([1] nodes joined)[{0f0e97a8c9e6}{QQZgyu48SHWMiyIYVxn8oQ}{UfllLjcrQmytunA-PUtM1A}{192.168.32.3}{192.168.32.3:9300}{dilm}{ml.machine_memory=67553959936, xpack.installed=true, ml.max_open_jobs=20} elect leader, _BECOME_MASTER_TASK_, _FINISH_ELECTION_], term: 22, version: 229, delta: master node changed {previous [], current [{0f0e97a8c9e6}{QQZgyu48SHWMiyIYVxn8oQ}{UfllLjcrQmytunA-PUtM1A}{192.168.32.3}{192.168.32.3:9300}{dilm}{ml.machine_memory=67553959936, xpack.installed=true, ml.max_open_jobs=20}]}" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-25T15:48:52,937Z", "level": "INFO", "component": "o.e.c.s.ClusterApplierService", "cluster.name": "docker-cluster", "node.name": "0f0e97a8c9e6", "message": "master node changed {previous [], current [{0f0e97a8c9e6}{QQZgyu48SHWMiyIYVxn8oQ}{UfllLjcrQmytunA-PUtM1A}{192.168.32.3}{192.168.32.3:9300}{dilm}{ml.machine_memory=67553959936, xpack.installed=true, ml.max_open_jobs=20}]}, term: 22, version: 229, reason: Publication{term=22, version=229}" }
zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-init_1           | FATAL:  role "zammad" does not exist
zammad-init_1           | Couldn't create 'zammad_production' database. Please check your configuration.
zammad-init_1           | rake aborted!
zammad-init_1           | ActiveRecord::NoDatabaseError: FATAL:  role "zammad" does not exist
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:696:in `rescue in connect'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:691:in `connect'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:223:in `initialize'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `new'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `postgresql_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:830:in `new_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:874:in `checkout_new_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:853:in `try_to_checkout_new_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:814:in `acquire_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:538:in `checkout'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:1033:in `retrieve_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_handling.rb:90:in `connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/postgresql_database_tasks.rb:12:in `connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/postgresql_database_tasks.rb:21:in `create'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:119:in `create'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:139:in `block in create_current'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:316:in `block in each_current_configuration'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:313:in `each'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:313:in `each_current_configuration'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:138:in `create_current'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/railties/databases.rake:29:in `block (2 levels) in <main>'
zammad-init_1           | /usr/local/bundle/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
zammad-init_1           | /usr/local/bin/bundle:30:in `block in <main>'
zammad-init_1           | /usr/local/bin/bundle:22:in `<main>'
zammad-init_1           | 
zammad-init_1           | Caused by:
zammad-init_1           | PG::ConnectionBad: FATAL:  role "zammad" does not exist
zammad-init_1           | /usr/local/bundle/gems/pg-0.21.0/lib/pg.rb:56:in `initialize'
zammad-init_1           | /usr/local/bundle/gems/pg-0.21.0/lib/pg.rb:56:in `new'
zammad-init_1           | /usr/local/bundle/gems/pg-0.21.0/lib/pg.rb:56:in `connect'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:692:in `connect'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:223:in `initialize'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `new'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `postgresql_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:830:in `new_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:874:in `checkout_new_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:853:in `try_to_checkout_new_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:814:in `acquire_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:538:in `checkout'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:1033:in `retrieve_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_handling.rb:90:in `connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/postgresql_database_tasks.rb:12:in `connection'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/postgresql_database_tasks.rb:21:in `create'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:119:in `create'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:139:in `block in create_current'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:316:in `block in each_current_configuration'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:313:in `each'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:313:in `each_current_configuration'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/tasks/database_tasks.rb:138:in `create_current'
zammad-init_1           | /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/railties/databases.rake:29:in `block (2 levels) in <main>'
zammad-init_1           | /usr/local/bundle/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
zammad-init_1           | /usr/local/bin/bundle:30:in `block in <main>'
zammad-init_1           | /usr/local/bin/bundle:22:in `<main>'
zammad-init_1           | Tasks: TOP => db:create
zammad-init_1           | (See full trace by running task with --trace)

```
### Steps to reproduce the behavior:

* I have changed version in `.env` file into VERSION = -3.2.0-13 (latest one)
* I have increased sysctl -w vm.max_map_count to 327680
* Command `docker-compose up`

**Summary**
I think that Elasticsearch container works here properly but there are problems with _zammad-init_ and _zammad-postgresql_.

Can you explain me that what `FATAL:  role "zammad" does not exist` in _zammad-postgresql_ and _zammad-init_ logs means?

Do you have any ideas how to fix these problems? I would be very grateful for your help!

**Comment by MrGeneration** [comment]
Duplicate of https://github.com/zammad/zammad-docker-compose/issues/132
Issue v1 actually depends on issue v2 as your railsserver never gets ready (due to missing postgres)

**Comment by Waasil** [comment]
Okey but I don't understand what do you mean that 
> railsserver never gets ready (due to missing postgres).

I follow every step in docs and there is always an issue which I have described in **Problem_v2**.
Does it mean that zammad-postgresql is invalid?

**Comment by MrGeneration** [comment]
Your railsserver depends on the postgresql container to become ready, before the railsserver container can get ready. These two containers must be ready before the elasticsearch container can start doing its job.

It's actually telling you within it's log as well.
So basically, you only have one problem which is postgres not starting - the elasticsearch issue is technically no real issue, because your dependencies simply are not ready to go.


---


I classified this issue as duplicate, as the linked issue is exactly about the same error message. Thus I'm closing the duplicates because it doesn't make sense to keep them open.

Please patiently wait until there's a response on the linked issue.
